app/assets/javascripts/ephemera_project_slug.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Ephemera Project slug generation
+ * Generata slug identifier when the client enters or modifies the title for a project
+ */
+
+// UUID Generation without Bower, AMD's, or the NPM
+function uuidv4() {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
```

Comment:
does this become the ID of project, somehow?

Reply:
Thank you for the review, documentation should indeed be added in order to clarify this feature.

> Shouldn't there be a routing piece? What happens if a slug changes? Is there some 'preferred' or 'primary' slug [...]

This does not become the ID of the EphemeraProject.  As the slug is entered by the user prior to ingestion (when the ID is assigned), this is simply a UUID-based alternative for generating unique slugs for EphemeraProject Objects.

A unique slug is required for the purposes of publishing messages via the RabbitMQ, but I don't believe that users ever actually reference or otherwise use this slug (from what I've gathered this is used for synchronization with other services such as https://github.com/pulibrary/lae-blacklight).  Hence, I don't believe that routing or redirection should be offered.  Please correct me if I am incorrect @tpendragon or @escowles.

Unless there is a preference for users to generate the slug using their own scheme (i. e. if they happened to track these slugs for purposes of which I'm unaware), then I would think that the automatic generation using JavaScript is entirely unwarranted.

Reply:
The slug is the user-readable shortname for the project/collection — e.g., for use in DPUL, where the collection URL is dpul.princeton.edu/[slug].  So it should be entered by the user, and shouldn't be auto-generated or have any suffixes, etc. to make sure it's unique.  If there is another collection or ephemera project with the slug, an we should just display an error message to the user.

Reply:
Ah, okay, my understanding was that this would be the basis of a distributable URL to aid in replacing standalone sites like the one for lae with easy-to-access collection pages within figgy/plum. Was I mistaken here? @escowles? Totally possible, considering I didn't know about the messaging use case.

Reply:
@hackmastera I think that's right, except the collection pages are in DPUL, not figgy/plum.  The point of the slug property is for plum/figgy to tell DPUL what it should use for the collection shortname for building its URLs.

Reply:
Okay, but we don't need routing because it's not really for use in figgy / plum -- it's just for use in DPUL.